services/markup/processorhelper.go

```diff
@@ -0,0 +1,29 @@
+// Copyright 2022 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package markup
+
+import (
+	"context"
+
+	"code.gitea.io/gitea/models/user"
+	"code.gitea.io/gitea/modules/log"
+	"code.gitea.io/gitea/modules/markup"
+)
+
+func ProcessorHelper() *markup.ProcessorHelper {
+	return &markup.ProcessorHelper{
+		IsUsernameMentionable: func(ctx context.Context, username string) bool {
+			// TODO: cast ctx to modules/context.Context and use IsUserVisibleToViewer
```

Comment:
This should be really fixed as otherwise it's security issue

Reply:
Yep, strongly agree.

Reply:
I'll give it a shot

Reply:
Wouldn't I be able to confirm a user exists by trying to sign up with their username?

Reply:
> This should be really fixed as otherwise it's security issue

Yup,  so the TODO is kept here (I would take it if it's needed), the fix needs other works.

Reply:
> Wouldn't I be able to confirm a user exists by trying to sign up with their username?

It should not as both invalid username and invalid password should result in same error message

Reply:
Here's how I confirm there is a private user named `hidden`
![image](https://user-images.githubusercontent.com/20454870/197358250-d35d8319-6602-47df-ab94-2d379ff53080.png)

Reply:
For public instances you can but for private you won't be able to do that as signup form will be disabled

Reply:
Gotcha

Reply:
My 2 cents, for a private instance, as a company's instance, the users (staffs) could know all others' names. The staff usernames are quite regular.

(I don't mean that it is not a security issue, but it's not that serious in many cases 😁 )

Reply:
Still many companies have external contractors and users that are restricted and should have access to only needed information

Reply:
The next PR is
* https://github.com/go-gitea/gitea/pull/21554

It LGTM